src/components/page_components/study_plans/StudyPlanTable.tsx

```diff
@@ -0,0 +1,130 @@
+import { ActionIcon, Checkbox, Table, TextInput, Tooltip } from "@mantine/core";
+import { useDebouncedState, useListState } from "@mantine/hooks";
+import { StudyPlanElement } from "@ourtypes/study_plans_page_types/StudyPlanElement";
+import { useState } from "react";
+import { FaDownload, FaEye, FaSearch } from "react-icons/fa";
+import { StudyPlanTableRowsTemplate } from "./StudyPlanTableRowsTemplate";
+
+interface StudyPlanTableProps {
+	elements: StudyPlanElement[];
+}
+
+// XD
+async function downloadFile(element: StudyPlanElement) {
+	const fileUrl = `https://center-be.stamford.dev/api/get_files/${element.fileKey}?download=true`;
+
+	try {
+		const a = document.createElement("a");
+		a.href = fileUrl;
+		a.target = "_blank";
+		a.download = `${element.major}_${element.year}_${element.language}.pdf`;
+		a.style.display = "none"; // Hide the element from view
+
+		document.body.appendChild(a);
+		a.click();
+		document.body.removeChild(a); // Remove the element from the DOM
+	} catch (error) {
+		console.error(error);
```

Comment:
It opens a new tab and will only download if the content-disposition and all that is set, there's no error returned to the client since I didn't make it parse the result